File: matiasmasso/Mat.NET/DT2/DTOGrf.vb
Public Class DTOGrf
    Property FchFrom As Date
    Property FchTo As Date
    Property DateInterval As DateInterval
    Property Items As List(Of DTOGrfItem)

End Class

Public Class DTOGrfItem
    Property Fch As Date
    Property Value As Decimal
End Class

Public Class DTOGrfMesValue

    Property Product As DTOProduct
    Property Color As Color
    Property Mesos As List(Of DTOYearMonth)
    Property Resto As Decimal

    Public Enum Codis
        Comandes
        Albarans
    End Enum

    Function Sum() As Decimal
        Dim retval As Decimal = _Mesos.Sum(Function(x) x.eur)
        Return retval
    End Function


    Public Sub New(oProduct As DTOProduct, ByVal oYearMonths As List(Of DTOYearMonth))
        _Product = oProduct
        _Mesos = oYearMonths
    End Sub

End Class

File: matiasmasso/Mat.NET/DTO/DTOGrf.cs
using DTO;
using MatHelperStd;
using System.Drawing;
using System;
using System.Collections.Generic;
using System.Linq;
public class DTOGrf
{
    public DateTime FchFrom { get; set; }
    public DateTime FchTo { get; set; }
    public TimeHelper.DateIntervals DateInterval { get; set; }
    public List<DTOGrfItem> Items { get; set; }
}

public class DTOGrfItem
{
    public DateTime Fch { get; set; }
    public decimal Value { get; set; }
}

public class DTOGrfMesValue
{
    public DTOProduct Product { get; set; }
    public Color Color { get; set; }
    public List<DTOYearMonth> Mesos { get; set; }
    public decimal Resto { get; set; }

    public enum Codis
    {
        Comandes,
        Albarans
    }

    public decimal Sum()
    {
        decimal retval = Mesos.Sum(x => x.Eur);
        return retval;
    }


    public DTOGrfMesValue(DTOProduct oProduct, List<DTOYearMonth> oYearMonths)
    {
        Product = oProduct;
        Mesos = oYearMonths;
    }
}
